File: ASP.NET + Angular/Client/src/app/models/post.ts
import { User } from "./user";
import { Comment } from "./comment";
import { UserLike } from "./userLike";
import { UserSave } from "./userSave";

export interface Post {
    id: number;
    imageUrl: string;
    description: string;
    createdBy: string;
    isDeleted: boolean;
    user: User;
    comments: Array<Comment>;
    likes: Array<UserLike>;
    saves: Array<UserSave>;
}

File: ASP.NET + Angular/Instagram/Data/Entities/Post.cs
using Data.Entities.Base;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Data.Entities
{
    public class Post : DeletableEntity
    {
        public int Id { get; set; }

        [Required]
        public string ImageUrl { get; set; }

        [Required]
        public string Description { get; set; }

        [Required]
        public string UserId { get; set; }

        public User User { get; set; }

        public List<UserLike> Likes { get; set; }

        public List<UserSave> Saves { get; set; }

        public List<Comment> Comments { get; set; }
    }
}
